Revit_Core_Engine/Convert/Geometry/ToRevit/XYZ.cs

```diff
@@ -47,9 +47,17 @@ public static XYZ ToRevit(this oM.Geometry.Point point)
         [Output("xyz", "Revit XYZ resulting from converting the input BH.oM.Geometry.Vector.")]
         public static XYZ ToRevit(this oM.Geometry.Vector vector)
         {
-            return new XYZ(vector.X.FromSI(SpecTypeId.Length), vector.Y.FromSI(SpecTypeId.Length), vector.Z.FromSI(SpecTypeId.Length));
+            return new XYZ(vector.X * m_LengthFromSi, vector.Y * m_LengthFromSi, vector.Z * m_LengthFromSi);
         }
 
+
+        /***************************************************/
+        /****              Private fields               ****/
+        /***************************************************/
+
+        // Optimisation to avoid calling FromSI(SpecTypeId.Length) every time as it causes massive overhead
+        private static double m_LengthFromSi = 1.0.FromSI(SpecTypeId.Length);
```

Comment:
Would look more consistent...
```suggestion
        private static double m_LengthFromSI = 1.0.FromSI(SpecTypeId.Length);
```